### ISSUE TITLE
ImportError: cannot import name 'RecursiveCharacterTextSplitter' from 'langchain.text_splitter'

### ISSUE THREAD

**Issue by dPacc** [hwchase17/chat-your-data#6]
Hi there, I'm trying to run this locally, but the following error occurs when I run `python ingest_data.py`

**Comment by lightshifted** [comment]
I encountered a similar issue solved by:
`pip install --upgrade langchain`

Hopefully this works for you as well!

**Comment by akiral** [comment]
Also confirm that you're running the right version of Python. 

`py --version`

**Comment by AcerScottLee** [comment]
> Also make sure that you are using the correct Python version: > > `py --version`.

Which of the Python versions is required?  I use Python 3.8 but there will be this error, TKS

**Comment by anantdevcs** [comment]
any update on this?

**Comment by ZXXSG** [comment]
> Hi there, I'm trying to run this locally, but the following error occurs when I run `python ingest_data.py`

 I meet the same problem.  How did you solve it in the end?

**Comment by AcerScottLee** [comment]
Have you ever tried running "pip install --upgrade langchain"?

**Comment by AcerScottLee** [comment]
> > Hi there, I'm trying to run this locally, but the following error occurs when I run `python ingest_data.py`您好，我正在嘗試在本地運行它，但是當我運行 `python ingest_data.py` 時出現以下錯誤
> 
> I meet the same problem. How did you solve it in the end?我遇到了同樣的問題。你最後是怎麼解決的？

Have you ever tried running "pip install --upgrade langchain"?

**Comment by lsh1215** [comment]
Of course, it won't be helpful much later, but I found a solution. It seems to be a version issue. Currently, instead of importing from langchain.text_splitter, you should import from langchain_text_splitters.

You can install it via:
`pip install -qU langchain-text-splitters`

And then modify your import statement to:
`from langchain_text_splitters import RecursiveCharacterTextSplitter`

Try this and see if it resolves the issue.

https://python.langchain.com/v0.1/docs/modules/data_connection/document_transformers/recursive_text_splitter/

**Comment by shabh2412** [comment]
> Of course, it won't be helpful much later, but I found a solution. It seems to be a version issue. Currently, instead of importing from langchain.text_splitter, you should import from langchain_text_splitters.
> 
> You can install it via: `pip install -qU langchain-text-splitters`
> 
> And then modify your import statement to: `from langchain_text_splitters import RecursiveCharacterTextSplitter`
> 
> Try this and see if it resolves the issue.
> 
> https://python.langchain.com/v0.1/docs/modules/data_connection/document_transformers/recursive_text_splitter/

This worked for me, thanks for sharing!

**Comment by LukeKvamme** [comment]
> Of course, it won't be helpful much later, but I found a solution. It seems to be a version issue. Currently, instead of importing from langchain.text_splitter, you should import from langchain_text_splitters.
> 
> You can install it via: `pip install -qU langchain-text-splitters`
> 
> And then modify your import statement to: `from langchain_text_splitters import RecursiveCharacterTextSplitter`
> 
> Try this and see if it resolves the issue.
> 
> https://python.langchain.com/v0.1/docs/modules/data_connection/document_transformers/recursive_text_splitter/

Worked perfectly even a few years later, thank you!
